app/src/main/java/org/mozilla/fenix/settings/logins/LoginDetailFragment.kt

```diff
@@ -0,0 +1,243 @@
+import kotlinx.coroutines.*
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.Dispatchers.Main
+import mozilla.components.concept.storage.Login
+import mozilla.components.lib.state.ext.consumeFrom
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.FenixSnackbar
+import org.mozilla.fenix.components.StoreProvider
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.showToolbar
+
+/**
+ * Displays saved login information for a single website.
+ */
+@ExperimentalCoroutinesApi
+class LoginDetailFragment : Fragment(R.layout.fragment_login_detail) {
+
+    private val args by navArgs<LoginDetailFragmentArgs>()
+    private var login: SavedLogin? = null
+    private lateinit var savedLoginsStore: LoginsFragmentStore
+    private lateinit var loginDetailView: LoginDetailView
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_login_detail, container, false)
+        savedLoginsStore = StoreProvider.get(this) {
+            LoginsFragmentStore(
+                LoginsListState(
+                    isLoading = true,
+                    loginList = listOf(),
+                    filteredItems = listOf()
+                )
+            )
+        }
+        loginDetailView = LoginDetailView(view?.findViewById(R.id.loginDetailLayout))
+        fetchLoginDetails()
+
+        return view
+    }
+
+    @ObsoleteCoroutinesApi
+    @ExperimentalCoroutinesApi
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        consumeFrom(savedLoginsStore) {
+            loginDetailView.update(it)
+            login = savedLoginsStore.state.currentItem
+//            showToolbar(savedLoginsStore.state.currentItem?.origin ?: "")
+            setUpCopyButtons()
+            showToolbar(savedLoginsStore.state.currentItem?.origin ?: "")
+            setUpPasswordReveal()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
+        super.onCreate(savedInstanceState)
+    }
+
+    private fun setUpPasswordReveal() {
+        passwordInfoText.inputType =
+            InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+        revealPasswordButton.setOnClickListener {
+            togglePasswordReveal()
+        }
+    }
+
+    private fun setUpCopyButtons() {
+        val copyInfoArgs = listOf(
+            Pair(copyWebAddress, login?.origin),
+            Pair(copyUsername, login?.username),
+            Pair(copyPassword, login?.password)
+        )
+
+        for (copyButtons in copyInfoArgs) {
+            copyButtons.first.setOnClickListener(
+                CopyButtonListener(
+                    copyButtons.second,
+                    R.string.login_copied
```

Comment:
These all are supposed to have the same message displayed when you copy them? Shouldn't it be specific to what has been copied? 🤔